app/controllers/schools/add_participants_controller.rb

```diff
@@ -5,7 +5,7 @@ class AddParticipantsController < ::Schools::BaseController
     include Multistep::Controller
 
     skip_after_action :verify_authorized
-    before_action :set_school_cohort
+    prepend_before_action :set_school_cohort
```

Comment:
What's the difference?

Reply:
So that's the bit I am not very happy with - there is a before_action defined in `Multiform::Controller` that checks the sesssion is in correct state and redirects to the `abandon_journey_path` it is not. `abandon_journey_path` requires @school_cohort variable to be set to work - and this one is set in `:set_school_cohort`. Sooo, we need `set_school_cohort` to run before the `before_action` from `Multiform::Controller` and prepend is enforcing this.

Now, what's odd is that rails have `append_before_action` which I'd love to use for `ensure_form_present` check, the problem is that it is just an alias for `before_action` - rails is storing all the filters in a single array, so you cannot force for the filter to be last.

Reply:
Oh, actually one potential fix to this would be to move this redirection from before_action to the top of show and update actions. This is much better as it will allow other actions to be defined on the controller without triggering the filter.

Reply:
Well, that's a bit unfortunate. We could (I guess) decouple `abandon_journey_path` from the `set_school_cohort` setup, but, ehhh, I am not sure I like it.

I will leave it up to your judgement :)

Reply:
Oooh, your solution sounds much better. Let's go for that!